### PR TITLE
🐛 Source Quickbooks: Replace Custom Components with Airbyte CDK

### LINKED ISSUE #44367: [source-quickbooks] Trying to set-up quickbooks
### Connector Name

destination-quickbooks

### Connector Version

3.0.17

### What step the error happened?

Configuring a new connector

### Relevant information

System: Windows
Version: Community
Fresh install of Airbyte and Docker.

Steps taken:
1. Installed Docker
2. Installed Airbyte
3. Logged In
4. Selected New Source
5. Selected Marketplace
6. Filled out settings

Received this error: 
"Unable to connect to stream accounts - 'str' object has no attribute 'eval'"

Logs pasted below.

Any advice appreciated!

### Relevant log output

```shell
2024-08-17 20:53:42 [32mINFO[m i.a.w.l.p.s.m.Stage(apply):39 - APPLY Stage: CLAIM — (workloadId = cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_0484d460-3449-4070-bfdc-40cf509bac4f_0_check) — (dataplaneId = local)
2024-08-17 20:54:05 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 - 
2024-08-17 20:54:06 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 - ----- START CHECK -----
2024-08-17 20:54:06 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 - 
2024-08-17 20:54:20 [32mINFO[m i.a.c.ConnectorWatcher(run):87 - Connector exited, processing output
2024-08-17 20:54:20 [32mINFO[m i.a.c.ConnectorWatcher(run):90 - Output file jobOutput.json found
2024-08-17 20:54:20 [32mINFO[m i.a.c.ConnectorWatcher(run):96 - Connector exited with 0
2024-08-17 20:54:20 [32mINFO[m i.a.w.i.VersionedAirbyteStreamFactory(create):189 - Reading messages from protocol version 0.2.0
2024-08-17 20:54:20 [33mWARN[m i.a.m.l.MetricClientFactory(getMetricClient):43 - MetricClient has not been initialized. Must call MetricClientFactory.CreateMetricClient before using MetricClient. Using a dummy client for now. Ignore this if Airbyte is configured to not publish any metrics.
2024-08-17 20:54:21 [1;31mERROR[m i.a.w.i.VersionedAirbyteStreamFactory(internalLog):304 - Encountered an error trying to connect to stream accounts. Error: 
 Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/checks/check_stream.py", line 42, in check_connection
    stream_is_available, reason = availability_strategy.check_availability(stream, logger, source)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/streams/http/availability_strategy.py", line 50, in check_availability
    get_first_record_for_slice(stream, stream_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/streams/utils/stream_helper.py", line 40, in get_first_record_for_slice
    return next(records_for_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/declarative_stream.py", line 126, in read_records
    yield from self.retriever.read_records(self.get_json_schema(), stream_slice)
  File "/usr/local/lib/python3.10/site-packages/airbyte_cdk/sources/declarative/retrievers/simple_retriever.py", line 339, in read_records
    self.cursor.close_slice(_slice, most_recent_record_from_slice)
  File "/airbyte/integration_code/source_quickbooks/components.py", line 72, in close_slice
    most_recent_record=LastRecordDictProxy(most_recent_record, {self.cursor_field.eval(self.config): "MetaData/LastUpdatedTime"}),
AttributeError: 'str' object has no attribute 'eval'

2024-08-17 20:54:21 [1;31mERROR[m i.a.w.i.VersionedAirbyteStreamFactory(internalLog):304 - Check failed
2024-08-17 20:54:21 [32mINFO[m i.a.c.ConnectorMessageProcessor(updateConfigFromControlMessage):231 - Checking for optional control message...
2024-08-17 20:54:21 [32mINFO[m i.a.c.ConnectorMessageProcessor(updateConfigFromControlMessage):253 - Optional control message not found. Skipping...
2024-08-17 20:54:21 [32mINFO[m i.a.c.ConnectorWatcher(run):134 - Writing output of cf9c4355-b171-4477-8f2d-6c5cc5fc8b7e_0484d460-3449-4070-bfdc-40cf509bac4f_0_check to the doc store
2024-08-17 20:54:21 [32mINFO[m i.a.c.ConnectorWatcher(run):136 - Marking workload as successful
2024-08-17 20:54:21 [32mINFO[m i.a.c.ConnectorWatcher(exitProperly):189 - Deliberately exiting process with code 0.
2024-08-17 20:54:21 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 - 
2024-08-17 20:54:21 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 - ----- END CHECK -----
2024-08-17 20:54:21 [32mINFO[m i.a.c.i.LineGobbler(voidCall):166 -
```

### Contribute

- [ ] Yes, I want to contribute

### PR DESCRIPTION
**Pull request by pabloescoder**
## What

The manipulations that we were doing in the custom components.py file to create a CustomDateTimeCursor were unnecessary (they were done to access the nested date time field for the cursor back when we had no support for transformations natively in the cdk)
However even after adding the transformation to bring the datetime value to the root level, we still kept the custom manipulations that caused this bug: https://github.com/airbytehq/airbyte/issues/44367

This PR scraps the custom logic and uses native Airbyte CDK features instead. The connector streams work now without any issues.
Resolves https://github.com/airbytehq/airbyte/issues/44367

## How

Use Airbyte CDK features to replace components.py.

## Review guide

1. `manifest.yaml`
 
## User Impact

Fixes bug that prevented the streams from working & fetching data.

## Can this PR be safely reverted and rolled back?

- [X] YES 💚

### CODE REVIEW

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/__init__.py [hunk 1]
```diff
@@ -3,7 +3,6 @@
 #
 
 
-from .components import CustomDatetimeBasedCursor
 from .source import SourceQuickbooks
 
-__all__ = ["SourceQuickbooks", "CustomDatetimeBasedCursor"]
+__all__ = ["SourceQuickbooks"]
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-quickbooks/source_quickbooks/__init__.py:8]: I think you can just remove the line altogether, and make this file empty. The file is still required, but I think no content is needed.

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml [hunk 2]
```diff
@@ -1,211 +1,1871 @@
-version: "1.0.0"
+version: 1.0.0
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml:1]: Let's bump this to 4.3.2 while we're here.

**Comment by natikgadzhi** [comment]
Regression tests run: https://github.com/airbytehq/airbyte/actions/runs/10515441620

**Review by natikgadzhi — COMMENTED** [review]
I've kicked off the regression tests. 

The change in the manifest looks brutally large — I think that' because Builder reformatted it. We _should_ be ok, but it's unreviewable.

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml (hunk 2 above)
Comment (pabloescoder) [reply airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml:1]: We'll have to bump the cdk version too then in poetry, is that fine?

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/__init__.py (hunk 1 above)
Comment (pabloescoder) [reply airbyte-integrations/connectors/source-quickbooks/source_quickbooks/__init__.py:8]: Sure, I'll do that

**Comment by pabloescoder** [comment]
> I've kicked off the regression tests.
> 
> The change in the manifest looks brutally large — I think that' because Builder reformatted it. We _should_ be ok, but it's unreviewable.

Just noticed that :( 
If it helps - I made changes in the query request param for each stream (it used $parameters before) and added a record selector to every stream (each one has a different path based on the stream name), those were the major changes, apart from that did some minor changes to remove $parameters

File: airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml (hunk 2 above)
Comment (pabloescoder) [reply airbyte-integrations/connectors/source-quickbooks/source_quickbooks/manifest.yaml]: Done, manifest version is 4.3.2 now :)

**Comment by natikgadzhi** [comment]
[Kicked-off regression tests](https://github.com/airbytehq/airbyte/actions/runs/10569716789)

**Comment by natikgadzhi** [comment]
@pabloescoder can you please resolve the conflicts and merge master in? I'll kick the regression tests again.

**Comment by pabloescoder** [comment]
@natikgadzhi Fixed the merge conflicts & merged master, we can run regression tests now once all the pipelines complete running.

**Comment by hakodexte** [comment]
[https://github.com/airbytehq/airbyte/issues/45911](url)
Linking this issue too as it seems to be related.

File: airbyte-integrations/connectors/source-quickbooks/pyproject.toml [hunk 3]
```diff
@@ -16,8 +16,8 @@ repository = "https://github.com/airbytehq/airbyte"
 include = "source_quickbooks"
 
 [tool.poetry.dependencies]
-python = "^3.9,<3.12"
-airbyte-cdk = "1.0.0"
+python = "^3.10,<3.12"
+airbyte-cdk = "^4"
```
Comment (natikgadzhi) [inline airbyte-integrations/connectors/source-quickbooks/pyproject.toml:20]: I wonder if we should jump to v5 next (separate PR)

**Review by natikgadzhi — APPROVED** [review]
Approving to kick-off progressive rollout!

**Pull request merged**